### ISSUE TITLE
[Feature Discussion] traffic shaping ?

### ISSUE THREAD

**Issue by DonZheng** [Nyr/openvpn-install#577]
Hi Nyr,

Can you please share some guidance on how to update your script to support traffic shaping?
At the moment, I share the same client.ovpn to multiple users and result in one user eating up most of the bandwidth. I would like to have a equal limit set up so everyone is sharing the bandwidth fairly.

Thanks in adv.

**Comment by DonZheng** [comment]
Hi Nyr,

Thanks for your quick response. Really appreciate it.

My users are not tech savvy so I shipped out client with ovpn pre-loaded. In my particular case, the security concern of sharing the cert is bearable, although no ideal.

If I put 'shaper 1000000' in the client.ovpn, it only affects that client's upload speed, right?

If I put 'shaper 1000000' in the server.conf, it will put the whole tun0 under 1mbps?

Is my understanding correct?

Thanks

**Comment by Nyr** [comment]
I don't think that you can use the `shaper` perameter at all, because my script sets up OpenVPN in server mode and OpenVPN supports traffic shaping natively in P2P mode only. At least that's my understanding, I haven't tried myself.

As already mentioned, your best bet is probably to assign static IPs to clients (difficult when they share the same certificate) and then implement traffic shaping for them with external tools.

It's going to be a bit difficult if you don't have experience, but doable if you want to learn. If not, I would suggest to contract someone.

**Comment by DonZheng** [comment]
Found this thread via Google, Is it something that is suitable for my case?

https://serverfault.com/questions/777875/how-to-do-traffic-shaping-rate-limiting-with-tc-per-openvpn-client

**Comment by Nyr** [comment]
Yes, a bit more complex than what you need, but looks good to me.

**Comment by DonZheng** [comment]
Thank you for your expertise. 💃
